### PR TITLE
Added new typing system

### PR DESCRIPTION
**Pull request by gustavoFreireS**
* added extended typing using rollup generator

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "name": "react-smart-tabs",
-  "version": "0.1.0",
+  "version": "0.2.5-alpha",
```
Comment (davidcostadev) [inline package.json]: Why this jump?

Comment (Mazuh) [reply package.json]: weren't you guys using `npm version` on master for publishing? :man_facepalming:

Comment (Mazuh) [reply package.json]: anyway, created https://github.com/EvoluxBR/react-smart-tabs/issues/45 for dealing with this later

Comment (gustavoFreireS) [reply package.json]: the thing here is I tested  4 "failed alpha releases" attempting to get types coming inside the npm package (and proper set up)

**Pull request merged**